ngi_reports/utils/entities.py

```diff
@@ -397,47 +400,111 @@ def populate_element_flowcell(self, log, **kwargs):
             laneObj.weighted_avg_qval_proj /= laneObj.total_reads_with_qval_proj
             laneObj.weighted_avg_qval_proj = round(laneObj.weighted_avg_qval_proj, 2)
 
-    def populate_ont_flowcell(self):
-        final_acquisition = self.fc_details.get("acquisitions")[-1]
-
-        if "_PA" in self.run_name or "_PB" in self.run_name:
-            self.type = "PromethION"
-        elif "_MN" in self.run_name:
-            self.type = "MinION"
-        fc_runparameters = self.fc_details.get("protocol_run_info", {})
-
-        self.fc_type = fc_runparameters.get("flow_cell").get(
-            "user_specified_product_code"
-        )  # product_code not specified for minion
-        run_arguments = fc_runparameters.get("args")
-        for arg in run_arguments:
-            if "min_qscore" in arg:
-                self.qual_threshold = float(arg.split("=")[-1])
-        self.n50 = float(
-            final_acquisition.get("read_length_histogram")[-1]
-            .get("plot")
-            .get("histogram_data")[0]
-            .get("n50")
-        )
-        self.total_reads = float(
-            final_acquisition.get("acquisition_run_info")
-            .get("yield_summary")
-            .get("read_count")
-        )
-        self.total_reads = round(self.total_reads / 1000000, 2)
+    def populate_ont_flowcell(self, log):
+        # TODO Handle no data
+        if self.fc_details.get("lims", {}) == {}:
+            log.warning(
+                f"Flowcell {self.run_name} has no LIMS information, please check and amend report manually"
+            )
+            self.exclude = True
+        else:
+            final_acquisition = self.fc_details.get("acquisitions")[-1]
+            if "_PA" in self.run_name or "_PB" in self.run_name:
+                self.type = "PromethION"
+            elif "_MN" in self.run_name:
+                self.type = "MinION"
+            fc_runparameters = self.fc_details.get("protocol_run_info", {})
+
+            self.fc_type = fc_runparameters.get("flow_cell").get(
+                "user_specified_product_code"
+            )  # product_code not specified for minion
+
+            self.fc_id = fc_runparameters.get("flow_cell").get(
+                "user_specified_flow_cell_id"
+            )
+            run_arguments = fc_runparameters.get("args")
+            for arg in run_arguments:
```

Comment:
If run_arguments is a list, we don't need to loop through it, we can just check `"min_qscore" in run_arguments`